src/thinkspace/client/thinkspace-builder-rat/addon/steps/content.coffee

```diff
@@ -107,8 +115,58 @@ export default step.extend
       ember.RSVP.all(promises).then (valids) =>
         resolve(!valids.contains(false))
 
-
   save: ->
     new ember.RSVP.Promise (resolve, reject) =>
       @validate().then (validity) =>
         resolve(validity)
+
+  add_question_item: (type) ->
+    manager = @get('manager')
+    @set_loading('irat')
+    manager.add_question_item(type).then (item) =>
+      @create_question_items({new_ids: [item.id]})
+      @reset_loading('irat')
+
+  delete_question_item: (type, item_obj) ->
+    manager = @get('manager')
+    manager.delete_question_item(type, item_obj).then =>
+      @create_question_items()
+
+  duplicate_question_item: (type, item_obj) ->
+    manager = @get('manager')
+    @set_loading('irat')
+    manager.duplicate_question_item(type, item_obj).then =>
+      @create_question_items()
+      @reset_loading('irat')
+
+  reorder_up: (type, item_obj) ->
+    manager = @get('manager')
+    @set_loading('irat')
+    manager.reorder_item(type, item_obj, -1).then =>
```

Comment:
Feels like all of these `manager.reorder_item` calls should be abstracted out.  Lots of repeat code in the `then` handler. Something like:

```
  @manager_reorder(type, obj, offset)
```